File: server/model/Photo.js

const mongoose = require('mongoose');

const photoSchema = new mongoose.Schema({
  url: { type: String, required: true },
  title: String,
  description: String,
  albumId: { type: mongoose.Schema.Types.ObjectId, ref: 'Album' },
  takenAt: Date,
  location: String,
  tags: [String],
  metadata: mongoose.Schema.Types.Mixed,
  favorite: { type: Boolean, default: false },
}, { timestamps: true });

module.exports = mongoose.model('Photo', photoSchema);

File: server/model/Photo.java

package server.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
public class Photo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String url;
    private String title;
    private String description;
    private Long albumId;
    private String location;
    private LocalDateTime takenAt;
    
    // Getters and setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getUrl() { return url; }
    public void setUrl(String url) { this.url = url; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public Long getAlbumId() { return albumId; }
    public void setAlbumId(Long albumId) { this.albumId = albumId; }
    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }
    public LocalDateTime getTakenAt() { return takenAt; }
    public void setTakenAt(LocalDateTime takenAt) { this.takenAt = takenAt; }
}
